libs/payments/cart/src/lib/cart.service.spec.ts

```diff
@@ -478,6 +479,68 @@ describe('CartService', () => {
         mockUpdateCart
       );
     });
+
+    it('updates cart with coupon code', async () => {
+      const mockCart = ResultCartFactory();
+      const mockUpdateCart = UpdateCartFactory({
+        couponCode: faker.word.noun(),
+      });
+      const mockPrice = StripePriceFactory();
+
+      jest.spyOn(cartManager, 'fetchCartById').mockResolvedValue(mockCart);
+      jest
+        .spyOn(productConfigurationManager, 'retrieveStripePrice')
+        .mockResolvedValue(mockPrice);
+      jest
+        .spyOn(promotionCodeManager, 'assertValidPromotionCodeNameForPrice')
+        .mockResolvedValue(undefined);
+
+      jest.spyOn(cartManager, 'updateFreshCart').mockResolvedValue();
+
+      await cartService.updateCart(
+        mockCart.id,
+        mockCart.version,
+        mockUpdateCart
+      );
+
+      expect(cartManager.updateFreshCart).toHaveBeenCalledWith(
+        mockCart.id,
+        mockCart.version,
+        mockUpdateCart
+      );
+    });
+
+    it('throws if coupon is not valid', async () => {
+      const mockCouponCode = faker.word.noun();
+      const mockPrice = StripePriceFactory();
+      const mockCart = ResultCartFactory({
+        couponCode: mockCouponCode,
+      });
+      const mockUpdateCart = UpdateCartFactory({
+        couponCode: mockCouponCode,
+      });
+
+      jest.spyOn(cartManager, 'fetchCartById').mockResolvedValue(mockCart);
+      jest
+        .spyOn(productConfigurationManager, 'retrieveStripePrice')
+        .mockResolvedValue(mockPrice);
+      jest
+        .spyOn(promotionCodeManager, 'assertValidPromotionCodeNameForPrice')
+        .mockImplementation(() => {
+          throw new CouponErrorExpired();
+        });
+      jest.spyOn(cartManager, 'updateFreshCart').mockRejectedValue(undefined);
+
+      await expect(
+        cartService.updateCart(mockCart.id, mockCart.version, mockUpdateCart)
+      ).rejects.toBeInstanceOf(CouponErrorExpired);
+
+      expect(cartManager.updateFreshCart).not.toHaveBeenCalledWith(
+        mockCart.id,
+        mockCart.version,
+        mockUpdateCart
+      );
```

Comment:
nit. My expectation is for `cartManager.updateFreshCart` not be called at all. If so could this be updated to `not.toHaveBeenCalled`?

Using `not.toHaveBeenCalledWith` reads as if the test is expecting the method to be called, just not with the specified parameters.